Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -179,10 +197,13 @@ function flatMap(mapper)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.flatMap requires that |this| be an Object.");
 
-    if (!@isCallable(mapper))
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.